pkgs/by-name/so/sourcegit/package.nix

```diff
@@ -0,0 +1,112 @@
+{
+  lib,
+  stdenv,
+  buildDotnetModule,
+  fetchFromGitHub,
+  dotnetCorePackages,
+  copyDesktopItems,
+  makeDesktopItem,
+  libicns,
+
+  libXcursor,
+  libXext,
+  libXi,
+  libXrandr,
+
+  git,
+  xdg-utils,
+}:
+
+buildDotnetModule (finalAttrs: {
+  pname = "sourcegit";
+  version = "2025.01";
+
+  src = fetchFromGitHub {
+    owner = "sourcegit-scm";
+    repo = "sourcegit";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-6FO0WRUcWm7xnuw6Br97+cWZhvzIOesg/eANZzeyxZo=";
+  };
+
+  patches = [ ./fix-darwin-git-path.patch ];
+
+  dotnet-sdk = dotnetCorePackages.sdk_9_0;
+  dotnet-runtime = dotnetCorePackages.runtime_9_0;
+
+  nugetDeps = ./deps.json;
+
+  projectFile = [ "src/SourceGit.csproj" ];
+
+  executables = [ "SourceGit" ];
+
+  dotnetFlags = [
+    "-p:DisableUpdateDetection=true"
+    "-p:DisableAOT=true"
+  ];
+
+  nativeBuildInputs = lib.optionals stdenv.hostPlatform.isLinux [
+    copyDesktopItems
+    libicns
+  ];
+
+  # these are dlopen-ed at runtime
+  # libXi is needed for right-click support
+  # not sure about what the other ones are needed for, but I'll include them anyways
+  runtimeDeps = [
+    libXcursor
+    libXext
+    libXi
+    libXrandr
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "SourceGit";
+      exec = "SourceGit";
+      icon = "SourceGit";
+      desktopName = "SourceGit";
+      terminal = false;
+      comment = finalAttrs.meta.description;
+    })
+  ];
+
+  # add fallback binaries to use if the user doesn't already have them in their PATH
+  preInstall = ''
+    makeWrapperArgs+=(
+      --suffix PATH : ${
+        lib.makeBinPath [
+          git
+          xdg-utils
+        ]
+      }
+    )
+  '';
+
+  postInstall =
+    lib.optionalString stdenv.hostPlatform.isLinux ''
+      icns2png -x build/resources/App.icns
+      for f in App_*x32.png; do
```

Comment:
Why `App_*x32.png` here? Shouldn't it be `App_*.png`, else it'll only match `App_32x32.png` right?

Reply:
It looks like the command was silently failing, since I needed to use the `build/resources/app/App.icns`
(why does the command silently fail...)

After that, running `ls *.png` gives this:
```
App_128x128x32.png  App_32x32x32.png    App_64x64x32.png
App_256x256x32.png  App_512x512x32.png
```

pretty weird format, so I'll write a comment explaining the format